src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/components/ImportFile/parseFile.ts

```diff
@@ -106,6 +107,45 @@ function getDuplicateIds(
   return dups;
 }
 
+function filterExtraneousSampleIds(
+  rows: Record<string, string>[],
+  editableSampleIds: Set<string>,
+  exampleSampleIds: Set<string>
+) {
+  const extraneousUniqueSampleIds = new Set<string>();
+
+  const filteredRows = rows.filter((item) => {
+    const currentPID = item.currentPrivateID;
+    if (
+      !editableSampleIds.has(currentPID) &&
+      !exampleSampleIds.has(currentPID)
+    ) {
+      extraneousUniqueSampleIds.add(currentPID);
+    } else {
+      return item;
+    }
+  });
+  const extraneousSampleIds: string[] = [...extraneousUniqueSampleIds];
+  return { extraneousSampleIds, filteredRows };
+}
+
+export function warnMissingMetadata(
+  metadata: SampleEditMetadataWebform
+): Set<keyof SampleEditMetadataWebform> | null {
+  const missingMetadata = new Set<keyof SampleEditMetadataWebform>();
+  const ALWAYS_REQUIRED: Array<keyof SampleEditMetadataWebform> = [
+    "privateId",
```

Comment:
Isn't `privateId` not present in the Edit version? Should this be `currentPrivateID` instead?

Reply:
the name changes to `privateId` when it's in the actual webform table